### ISSUE TITLE
SIGBUS in LMDB access after resume

### ISSUE THREAD

**Issue by 7oud** [BVLC/caffe#2830]
When resume training "bvlc_reference_caffenet" model from "caffenet_train_iter_210000.solverstate", caffe crashed. 
Log info:
>
I0728 10:38:25.941831  8640 solver.cpp:229]     Train net output #0: loss = 6.91586 (* 1 = 6.91586 loss)
I0728 10:38:25.941848  8640 solver.cpp:486] Iteration 238460, lr = 0.0001
I0728 10:38:32.075881  8640 solver.cpp:214] Iteration 238480, loss = 6.91293
I0728 10:38:32.075923  8640 solver.cpp:229]     Train net output #0: loss = 6.91293 (* 1 = 6.91293 loss)
I0728 10:38:32.075939  8640 solver.cpp:486] Iteration 238480, lr = 0.0001
*** Aborted at 1438051127 (unix time) try "date -d @1438051127" if you are using GNU date ***
PC: @     0x7fd83ea51fb3 (unknown)
*** SIGBUS (@0x7f0f6be91000) received by PID 8640 (TID 0x7ed81e0cd700) from PID 1810436096; stack trace: ***
    @     0x7fd83ecd6cb0 (unknown)
    @     0x7fd83ea51fb3 (unknown)
    @     0x7fd83f19a4b0 (unknown)
    @     0x7fd83f19a61d (unknown)
    @     0x7fd83f8875ca caffe::db::LMDBCursor::value()
    @     0x7fd83f82e41a caffe::DataLayer<>::InternalThreadEntry()
    @     0x7fd835cb9ce9 (unknown)
    @     0x7fd83eccee9a start_thread
    @     0x7fd83e9fc31d (unknown)
Bus error (core dumped)

**Comment by 7oud** [comment]
environment info
GPU: GTX 780Ti with 3GB vram
batch size: 64

First time, the training stopped by disk space is not enough, because too many solverstate & caffemodel were generated, After deleting some solverstate & caffemodel files, I resume training from caffenet_train_iter_210000.solverstate, and caffe crashed when Iteration 238480

**Comment by longjon** [comment]
Seems like an odd place for a `SIGBUS`, though I wonder if something was left corrupted when you ran out of disk?

Can you provide steps that reproduce this in the latest `master`?
